### PR TITLE
fixed `QabElem`, improved printing of char. tables

### PR DESCRIPTION
**Pull request by ThomasBreuer**
- fixed `QabElem` for GAP cyclotomics:
  We have to use the cyclotomic fields that are cached in `abelian_closure(QQ)`,  otherwise the elements created by `QabElem` will not be compatible.

- fixed the printing of cyclotomics in Julia:
  The correct cyclotomic fields cause a different `string` format for elements.

- In `show` for character tables, if the irrational values are shown via
  symbolic names then use `*` to denote algebraic conjugacy in quadratic fields
  (relative names like `X` and `X*`).

- Use `"text/latex"` not `"text/html"` to specify that one wants LaTeX format.

### CODE REVIEW

File: src/Groups/group_characters.jl [hunk 1]
```diff
@@ -232,14 +233,54 @@ function Base.iterate(wi::WordsIterator, state::Int)
...
+    F = parent(val)
+    flag, N = Hecke.iscyclotomic_type(F)
+    flag || error("$val is not an element of a cyclotomic field")
+
+    # `string` yields an expression of the right structure,
+    # but not in terms of `root`
+    # and without curly brackets for subscripts and superscripts.
+    str = string(val)
+    str = replace(str, "*" => "")
+    str = replace(str, string(F.S) => root*"_{"*string(N)*"}")
```
Comment (fingolfin) [inline src/Groups/group_characters.jl]: Does this work, too? It'd be more readable, IMHO
```suggestion
    str = replace(str, string(F.S) => "$root_{$N}")
```

File: src/Groups/group_characters.jl [hunk 2]
```diff
@@ -261,31 +302,44 @@ function matrix_of_strings(tbl::GAPGroupCharacterTable; alphabet::String = "")
...
             valbar = complex_conjugate(val)
             if valbar != val
-              push!(legend, valbar => "\\overline{"*name*"}")
-#TODO: represent the unique Galois conjugate of `A` different from `A` by `A*`,
-#      and show both `A` and `A*` in the footer;
-#      for elements in quadratic fields, show also an expression in terms of
+              # The complex conjugate of a known name is shown relative to
+              # that name.
+              disp = as_sum_of_roots(valbar.data, root)
+              push!(legend, (valbar, "\\overline{"*name*"}", disp))
```
Comment (fingolfin) [inline src/Groups/group_characters.jl]: I am told in idiomatic Julia, string interpolation is preferred over string concatenation
```suggestion
              push!(legend, (valbar, "\\overline{$name}", disp))
```

File: test/Groups/group_characters.jl [hunk 3]
```diff
@@ -21,12 +21,12 @@
...
+  @test String(take!(io)) == "L2(11)\n\n  2  2  2  1  .  .  1   .   .\n  3  1  1  1  .  .  1   .   .\n  5  1  .  .  1  1  .   .   .\n 11  1  .  .  .  .  .   1   1\n                             \n    1a 2a 3a 5a 5b 6a 11a 11b\n 2P 1a 1a 3a 5b 5a 3a 11b 11a\n 3P 1a 2a 1a 5b 5a 2a 11a 11b\n 5P 1a 2a 3a 1a 1a 6a 11a 11b\n11P 1a 2a 3a 5a 5b 6a  1a  1a\n                             \n χ₁  1  1  1  1  1  1   1   1\n χ₂  5  1 -1  .  .  1   B   B̅\n χ₃  5  1 -1  .  .  1   B̅   B\n χ₄ 10 -2  1  .  .  1  -1  -1\n χ₅ 10  2  1  .  . -1  -1  -1\n χ₆ 11 -1 -1  1  1 -1   .   .\n χ₇ 12  .  .  A A*  .   1   1\n χ₈ 12  .  . A*  A  .   1   1\n\nA = -ζ₅³ - ζ₅² - 1\nA* = ζ₅³ + ζ₅²\nB = ζ₁₁⁹ + ζ₁₁⁵ + ζ₁₁⁴ + ζ₁₁³ + ζ₁₁\nB̅ = -ζ₁₁⁹ - ζ₁₁⁵ - ζ₁₁⁴ - ζ₁₁³ - ζ₁₁ - 1\n"
```
Comment (fingolfin) [inline test/Groups/group_characters.jl]: Can we perhaps at least format these as triple quoted strings, that'd be slightly nicer for a human to read? I.e. like this (untested)
```suggestion
  @test String(take!(io)) == 
    """
    L2(11)

      2  2  2  1  .  .  1   .   .
      3  1  1  1  .  .  1   .   .
      5  1  .  .  1  1  .   .   .
     11  1  .  .  .  .  .   1   1
                         
        1a 2a 3a 5a 5b 6a 11a 11b
     2P 1a 1a 3a 5b 5a 3a 11b 11a
     3P 1a 2a 1a 5b 5a 2a 11a 11b
     5P 1a 2a 3a 1a 1a 6a 11a 11b
    11P 1a 2a 3a 5a 5b 6a  1a  1a
                         
     χ₁  1  1  1  1  1  1   1   1
     χ₂  5  1 -1  .  .  1   B   B̅
     χ₃  5  1 -1  .  .  1   B̅   B
     χ₄ 10 -2  1  .  .  1  -1  -1
     χ₅ 10  2  1  .  . -1  -1  -1
     χ₆ 11 -1 -1  1  1 -1   .   .
     χ₇ 12  .  .  A A*  .   1   1
     χ₈ 12  .  . A*  A  .   1   1

    A = -ζ₅³ - ζ₅² - 1
    A* = ζ₅³ + ζ₅²
    B = ζ₁₁⁹ + ζ₁₁⁵ + ζ₁₁⁴ + ζ₁₁³ + ζ₁₁
    B̅ = -ζ₁₁⁹ - ζ₁₁⁵ - ζ₁₁⁴ - ζ₁₁³ - ζ₁₁ - 1
    """
```

File: src/Groups/group_characters.jl (hunk 1 above)
Comment (ThomasBreuer) [reply src/Groups/group_characters.jl]: No, one gets the error message that `root_` is not defined.
One could write `"$("$root")_{$N}"` but is it more readable?

Comment (thofma) [reply src/Groups/group_characters.jl]: `$(root)_{$N}`.

Comment (ThomasBreuer) [reply src/Groups/group_characters.jl]: Yes, `"$(root)_{$N}"` works. Thanks.

**Review by fingolfin — APPROVED** [review]
Nice!

**Pull request merged**